webpack.config.js

```diff
@@ -0,0 +1,412 @@
+                options: {
+                  sourceMap: isEnvProduction,
+                },
+              },
+            ].filter(Boolean),
+          },
+
+          // Worker loader
+          {
+            test: /worker\.js$/,
+            use: [
+              require.resolve('worker-loader'),
+              require.resolve('babel-loader'),
+            ],
+          },
+
+          // Process JS with Babel.
+          {
+            test: /\.(js|jsx)$/,
+            loader: require.resolve('babel-loader'),
+            options: {
+              customize: require.resolve(
+                'babel-preset-react-app/webpack-overrides'
+              ),
+              cacheDirectory: true,
+              cacheCompression: isEnvProduction,
+              compact: isEnvProduction,
+            },
+          },
+
+          // "file" loader makes sure assets end up in the `build` folder.
+          // When you `import` an asset, you get its filename.
+          // This loader don't uses a "test" so it will catch all modules
+          // that fall through the other loaders.
+          {
+            loader: require.resolve('file-loader'),
+            // Exclude `js` files to keep "css" loader working as it injects
+            // it's runtime that would otherwise processed through "file" loader.
+            // Also exclude `html` and `json` extensions so they get processed
+            // by webpacks internal loaders.
+            exclude: [/\.(js|jsx)$/, /\.html$/, /\.json$/],
+            options: {
+              name: 'static/media/[name].[hash:8].[ext]',
+            },
+          },
+          // ** STOP ** Are you adding a new loader?
+          // Make sure to add the new loader(s) before the "file" loader.
+        ],
+      },
+    ],
+  },
+
+  plugins: [
+    new HtmlWebpackPlugin({
+      inject: true,
+      template: PATHS.INDEX_HTML,
+    }),
+
+    // Makes some environment variables available in index.html.
+    // The public URL is available as %PUBLIC_URL% in index.html, e.g.:
+    // <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico">
+    // In production, it will be an empty string unless you specify "homepage"
+    // in `package.json`, in which case it will be the pathname of that URL.
+    new InterpolateHtmlPlugin(HtmlWebpackPlugin, {
+      NODE_ENV: process.env.NODE_ENV,
+      PUBLIC_URL: publicUrl,
+    }),
+
+    // Makes some environment variables available to the JS code, for example:
+    // if (process.env.NODE_ENV === 'production') { ... }.
+    // It is absolutely essential that NODE_ENV was set to production here.
+    // Otherwise React will be compiled in the very slow development mode.
+    new webpack.DefinePlugin({
+      'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV),
+      '__DEV__': process.env.NODE_ENV === 'development',
+      '__PROD__': process.env.NODE_ENV === 'production',
+      'process.env.BUILD_NUMBER': process.env.BUILD_NUMBER || 'dev',
```

Comment:
we can use existing flags here isEnvProduction & isEnvDevelopment

Reply:
thnx, fixed